src/rgw/rgw_rest_client.cc

```diff
@@ -702,7 +702,9 @@ int RGWRESTStreamRWRequest::do_send_prepare(RGWAccessKey *key, map<string, strin
   if (send_data) {
     set_outbl(*send_data);
     send_data_hint = true;
+    set_send_length(send_data->length());
```

Comment:
we should call `set_send_length` before `set_outbl`.

if send_data is not nullptr, outbl is nullptr(set_outbl is not called), content length will be 0. because `set_outbl` will swap `send_data` and `outbl`.

```
  void set_outbl(bufferlist& _outbl) {
    outbl.swap(_outbl);
  }
```